hikari/impl/entity_factory.py

```diff
@@ -2661,6 +2734,13 @@ def deserialize_modal_interaction(self, payload: data_binding.JSONObject) -> mod
         if message_payload := payload.get("message"):
             message = self.deserialize_message(message_payload)
 
+        authorizing_integration_owners = {
+            application_models.ApplicationIntegrationType(int(integration_type)): snowflakes.Snowflake(
+                int(integration_owner_id)
+            )
```

Comment:
I believe for snowflakes, you do not need to cast it to an int before hand (as Snowflake is a subclass of int)